pkg/target/target.go

```diff
@@ -508,6 +532,42 @@ func Summary(targets []*Target) fleet.BundleSummary {
 	return bundleSummary
 }
 
+// tplFuncMap returns a mapping of all of the functions from sprig but removes potentially dangerous operations
+func tplFuncMap() template.FuncMap {
+	f := sprig.TxtFuncMap()
+	delete(f, "env")
+	delete(f, "expandenv")
+	delete(f, "include")
+	delete(f, "tpl")
+
+	return f
+}
+
+func processTemplateValues(valuesMap map[string]interface{}, templateContext map[string]interface{}) (map[string]interface{}, error) {
+	var tplResult bytes.Buffer
+
+	valuesMapStr, err := k8syaml.Marshal(valuesMap)
+	if err != nil {
+		return nil, err
+	}
+
+	tpl, err := template.New("values").Funcs(tplFuncMap()).Option("missingkey=error").Parse(string(valuesMapStr))
```

Comment:
So I'm getting an error from the test suite when I started adding some longer lines to the test suite for variable substitution.

Here is the tested bundleYamlWithTemplate I was playing around with.

```
const bundleYamlWithTemplate = `namespace: default
helm:
  releaseName: labels
  values:
    clusterName: "{{ .ClusterLabels.name }}"
    fromAnnotation: "{{ .ClusterAnnotations.testAnnotation }}"
    clusterNamespace: "{{ .ClusterNamespace }}"
    fleetClusterName: "{{ .ClusterName }}"
    workingLongClusterName: kubernets.io/cluster/{{ index .ClusterLabels "p-w" }}
    brokenLongClusterName: kubernets.io/cluster/{{ index .ClusterLabels "not-working-for-some-reason" }}
    customStruct:
      - name: "{{ .Values.topLevel }}"
        key1: value1
        key2: value2
      - element2: "{{ .Values.nested.secondTier.thirdTier }}"
      - "element3_{{ .ClusterLabels.envType }}": "{{ .ClusterLabels.name }}"
    funcs:
      upper: "{{ .Values.topLevel | upper }}_test"
      join: '{{ .Values.list | join "," }}'
diff:
  comparePatches:
  - apiVersion: networking.k8s.io/v1
    kind: Ingress
    name: labels-fleetlabelsdemo
    namespace: default
    operations:
    - op: remove
      path: /spec/rules/0/host
`
```

And here is the error I'm getting from my `brokenLongClusterName: kubernets.io/cluster/{{ index .ClusterLabels "not-working-for-some-reason" }}` line.
```
--- FAIL: TestProcessTemplateValues (0.00s)
    /home/gate/src/fleet/pkg/target/target_test.go:179: error during label processing template: values:1: unclosed action
FAIL
```
I've done a bit of digging and it appears to have something to do with the k8syam.Marshal conversion to a string.  New lines are getting inserted into the line. So `brokenLongClusterName: kubernets.io/cluster/{{ index .ClusterLabels "not-working-for-some-reason" }}` will end up like `brokenLongClusterName: kubernets.io/cluster/{{ index .ClusterLabels "not-working-for-some-reason"\n }}`.  I have not been able to explain why the new lines are getting inserted.
```

Reply:
https://github.com/kubernetes-sigs/yaml/issues/68 has details about what is happening with lines over 80 chars.  I can confirm using go-yaml v3  instead of the k8s sigs yaml library resolves the issue.

Reply:
That's an interesting catch. I will update the library and see if that will fix it. However, fundamentally, this issue is raised by the fact that I serialize the entire document and re-parse it as template string. Looking back at it, I can't help it but feel a little bit unhappy about this approach because a carefully crafted template variable could fundamentally change the yaml structure of the templated output. Perhaps a better approach would be to iterate on each individual key values and template them as strings instead. WDYT?

Reply:
I really liked the idea of treating all of the values as a big template so you can start to do some fun if/then logic and not try to cram it all on one line.   But this 80 char line break in the yaml libraries breaks that.  My current use case is better variable substitute and not doing crazy logic inside the values so I'm happy with each line getting process separately like I'm doing in #507.

Reply:
Indeed. That's what convinced me to consider it. However, an alternative approach would be to define a dedicated field in fleet.yaml like `templatesValues:` and let the user define it as a multi-line string. This way it is implied that the entire structure is templated as a whole.

Reply:
Interestingly enough, I couldn't reproduce the new line behaviour on my local system (go 1.17.6) -- but I switched to goyamlv3 just to be safe. I also refactored the templating function to be a lot more "structure" safe, as in only templating within the context of a specific string rather than the entire object.